core/src/main/java/hudson/Functions.java

```diff
@@ -1082,29 +1078,55 @@ public static String getFooterURL() {
      * Like {@link #getSortedDescriptorsForGlobalConfig(Predicate)} but with a constant truth predicate, to include all descriptors.
      */
     public static Collection<Descriptor> getSortedDescriptorsForGlobalConfig() {
-        return getSortedDescriptorsForGlobalConfig(Predicates.<GlobalConfigurationCategory>alwaysTrue());
+        return getSortedDescriptorsForGlobalConfig(descriptor -> true);
     }
 
     /**
      * @deprecated This is rather meaningless.
      */
     @Deprecated
     public static Collection<Descriptor> getSortedDescriptorsForGlobalConfigNoSecurity() {
-        return getSortedDescriptorsForGlobalConfig(Predicates.not(GlobalSecurityConfiguration.FILTER));
+        return getSortedDescriptorsForGlobalConfigUnclassified();
     }
 
     /**
-     * Like {@link #getSortedDescriptorsForGlobalConfig(Predicate)} but for unclassified descriptors only.
+     * Descriptors in the global configuration form that users with {@link Jenkins#MANAGE} permission can configure.
+     *
      * @since 1.506
      */
     public static Collection<Descriptor> getSortedDescriptorsForGlobalConfigUnclassified() {
-        return getSortedDescriptorsForGlobalConfig(new Predicate<GlobalConfigurationCategory>() {
-            public boolean apply(GlobalConfigurationCategory cat) {
-                return cat instanceof GlobalConfigurationCategory.Unclassified;
+        return getSortedDescriptorsForGlobalConfig(d -> d.getCategory() instanceof GlobalConfigurationCategory.Unclassified && Jenkins.get().hasPermission(d.getRequiredGlobalConfigPagePermission()));
+    }
+
+    /**
+     * Descriptors shown in the global configuration form to users with {@link Jenkins#SYSTEM_READ} permission.
+     *
+     * @return
+     * @since TODO
+     */
+    @Restricted(NoExternalUse.class)
+    public static Collection<Descriptor> getSortedDescriptorsForGlobalConfigUnclassifiedReadable() {
+        return getSortedDescriptorsForGlobalConfig(d -> d.getCategory() instanceof GlobalConfigurationCategory.Unclassified && (
+                Jenkins.get().hasPermission(d.getRequiredGlobalConfigPagePermission()) || Jenkins.get().hasPermission(Jenkins.SYSTEM_READ)));
+    }
+
+    @Restricted(NoExternalUse.class)
+    public static void checkAnyPermission(AccessControlled ac, Permission[] permissions) {
+        if (permissions == null) {
+            return;
+        }
+
+        boolean failed = true;
+        for (Permission permission : permissions) {
+            if (ac.hasPermission(permission)) {
+                failed = false;
             }
-        });
+        }
+        if (failed) {
+            ac.checkPermission(permissions[0]);
+        }
```

Comment:
if you only want to see if any permission matches break the loop (or return) after the fist match.

the error here is looking not so good.  whilst the exception would show user is missing foobar permision they may only need wibble permission.  can we be explicit in the error that they are missing any of the permissions by throwing a AUthZ exception with a populated message?

Reply:
should be simpler now,
![image](https://user-images.githubusercontent.com/21194782/74825930-9ef33380-5302-11ea-9929-2d04d704c3fa.png)